drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/ConditionBuilder.java

```diff
@@ -0,0 +1,222 @@
+                                  conditionColumn.getOperator(),
+                                  ( Float ) realCellValue.getNumericValue() );
+
+        } else if ( type.equals( DataType.TYPE_NUMERIC_INTEGER ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  getIntegerValue() );
+
+        } else if ( type.equals( DataType.TYPE_NUMERIC_LONG ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  getLongValue() );
+
+        } else if ( type.equals( DataType.TYPE_NUMERIC_SHORT ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  getShortValue() );
+
+        } else if ( type.equals( DataType.TYPE_BOOLEAN ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  realCellValue.getBooleanValue() );
+        } else if ( type.equals( DataType.TYPE_DATE ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  realCellValue.getDateValue() );
+        } else if ( type.equals( DataType.TYPE_COMPARABLE ) ) {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  getStringValue() );
+        } else {
+            return new Condition( field,
+                                  getColumn(),
+                                  conditionColumn.getOperator(),
+                                  getStringValue() );
+        }
+    }
+
+    private String getStringValue() {
+        String stringValue = realCellValue.getStringValue();
+        if ( stringValue == null ) {
+            return null;
+        } else if ( stringValue.isEmpty() ) {
+            return null;
+        } else {
+            return stringValue;
+        }
+    }
+
+    private Column getColumn() {
+        return index.columns
+                .where( HasIndex.index().is( utils.getModel().getExpandedColumns().indexOf( conditionColumn ) ) )
+                .select().first();
+    }
+
+    private Short getShortValue() {
+        if ( realCellValue.getNumericValue() != null ) {
+            return ( Short ) realCellValue.getNumericValue();
+        } else if ( realCellValue.getStringValue() == null || realCellValue.getStringValue().isEmpty() ) {
+            return null;
+        } else {
+            return new Short( realCellValue.getStringValue() );
```

Comment:
Good point, better to keep it safe. I'll fix this in the next PRs if you are ok with that. Happens to be that I already had to change this bit, but didn't fix this issue. 
This is a copy paste of the old, just moved to another location, so there should be no regression.